File: KoRadio/UI/ko_radio_mobile/lib/models/company_job_assignment.dart
import 'package:json_annotation/json_annotation.dart';

import 'package:ko_radio_mobile/models/company_employee.dart';
import 'package:ko_radio_mobile/models/job.dart';

part 'company_job_assignment.g.dart';

@JsonSerializable()
class CompanyJobAssignment {
  int? companyJobId;
	int? companyEmployeeId;
	int? jobId;
	DateTime assignedAt;
  CompanyEmployee? companyEmployee;
  Job? job;

	CompanyJobAssignment(this.companyEmployeeId,this.jobId,this.assignedAt,);

	factory CompanyJobAssignment.fromJson(Map<String, dynamic> json) => _$CompanyJobAssignmentFromJson(json);

	Map<String, dynamic> toJson() => _$CompanyJobAssignmentToJson(this);
}

File: KoRadio/KoRadio.Services/Database/CompanyJobAssignment.cs
using System;
using System.Collections.Generic;

namespace KoRadio.Services.Database;

public partial class CompanyJobAssignment
{
    public int CompanyJobId { get; set; }

    public int? CompanyEmployeeId { get; set; }

    public int? JobId { get; set; }

    public DateTime AssignedAt { get; set; }

    public bool IsFinished { get; set; }
    public bool IsCancelled { get; set; }

	public virtual CompanyEmployee? CompanyEmployee { get; set; }

    public virtual Job? Job { get; set; }
}
